sos/policies/distros/uniontech.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (C) Red Hat, Inc. 2019
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.policies.distros.redhat import RedHatPolicy, OS_RELEASE
+import os
+
+
+class UnionTechPolicy(RedHatPolicy):
+    distro = "Union Tech Linux"
+    vendor = "The UnionTech Project"
+    vendor_urls = [('Distribution Website', 'https://www.chinauos.com/')]
+
+    def __init__(self, sysroot=None, init=None, probe_runtime=True,
+                 remote_exec=None):
+        super(UnionTechPolicy, self).__init__(sysroot=sysroot, init=init,
+                                              probe_runtime=probe_runtime,
+                                              remote_exec=remote_exec)
+
+    @classmethod
+    def check(cls, remote=''):
+
+        if remote:
+            return cls.distro in remote
+
+        if not os.path.exists(OS_RELEASE):
+            return False
+
+        with open(OS_RELEASE, 'r') as f:
+            for line in f:
+                if line.startswith('NAME'):
+                    if 'UnionTech' in line:
+                        return True
+        return False
+
```

Comment:
It looks as thought this policy was based on the Amazon policy (`sos/policies/distros/amazon.py`). In that case this works because Amazon's Linux distribution is based on and shares a lot of similarity with the Red Hat family of distributions. They have a common package manager (RPM), and similarities in file system layout that mean sharing policy makes sense.

If I understand correctly UOS/Union Tech Linux is based on Deepin, which itself is based on Debian Stable. If that is still the case then it probably makes more sense to base the Union Tech policy on one of the Debian family of distributions, since there will be greater commonality (of fs layout, packaging etc.) than with a Red Hat based distro.

Reply:
We have two versions. Debian is our desktop version system. I also have a server version. This time, the server version is supported. The server version is a secondary development version based on the openEuler system.